Source/Core/Common/Config/ConfigInfo.h

```diff
@@ -30,10 +34,30 @@ struct Location
 };
 
 template <typename T>
-struct Info
+struct CachedValue
 {
-  Info(const Location& location_, const T& default_value_)
-      : location{location_}, default_value{default_value_}
+  T value;
+  u64 config_version;
+};
+
+template <typename T>
+class Info
+{
+public:
+  constexpr Info(const Location& location, const T& default_value)
+      : m_location{location}, m_default_value{default_value}, m_cached_value{default_value, 0}
+  {
+  }
+
+  Info(const Info<T>& other)
+      : m_location{other.GetLocation()}, m_default_value{other.GetDefaultValue()},
+        m_cached_value{other.GetCachedValue()}
+  {
+  }
+
+  Info(Info<T>&& other)
```

Comment:
Ditto

```cpp
Info(Info<T>&& other) { *this = std::move(other); }
```

Reply:
Can't I leave out the `std::move` since `other` already is an xvalue?